### PR TITLE
Fix static tests to be php8 compatible

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Stabilisation of Static Tests to be compatible both with PHP7 and PHP8.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#34188

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: dev/tests/static/testsuite/Magento/Test/Integrity/Magento/Framework/Api/ExtensibleInterfacesTest.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ private function checkSetExtensionAttributes(
             } else {
                 // Get the parameter name via a regular expression capture because the class may
                 // not exist which causes a fatal error
-                preg_match('/\[\s\<\w+?>\s([\w]+)/s', $methodParameters[0]->__toString(), $matches);
+                preg_match('/\[\s\<\w+?>\s([?]?[\w]+)/s', $methodParameters[0]->__toString(), $matches);
```
Comment (karyna-t) [inline dev/tests/static/testsuite/Magento/Test/Integrity/Magento/Framework/Api/ExtensibleInterfacesTest.php]: **PHP 8**
Parameter #0 [ <required> ?Magento\MediaGalleryApi\Api\Data\KeywordExtensionInterface $extensionAttributes ]

**PHP 7**
Parameter #0 [ <required> Magento\MediaGalleryApi\Api\Data\KeywordExtensionInterface or NULL $extensionAttributes ]

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests with env PHP 8.0

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests

**Comment by andrewbess** [comment]
@magento run Static Tests with env PHP 8.0

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests with env PHP 8.0

**Comment by andrewbess** [comment]
@magento run all tests with env PHP 8.0

**Comment by andrewbess** [comment]
@magento run all tests

File: dev/tests/static/testsuite/Magento/Test/Integrity/Di/CompilerTest.php [hunk 2]
```diff
@@ -127,6 +127,7 @@ private function getPluginBlacklist(): array
             );
             $blacklistItems = [];
             foreach (glob($blacklistFiles) as $fileName) {
+                // phpcs:ignore Magento2.Performance.ForeachArrayMerge
```
Comment (xmav) [inline dev/tests/static/testsuite/Magento/Test/Integrity/Di/CompilerTest.php]: Could you please try to rewrite it using example from - https://github.com/magento/magento-coding-standard/issues/20

File: dev/tests/static/testsuite/Magento/Test/Integrity/Di/CompilerTest.php [hunk 3]
```diff
@@ -243,6 +244,7 @@ protected function _phpClassesDataProvider()
         $allowedFiles = array_keys($classes);
         foreach ($classes as $class) {
             if (!in_array($class, $output)) {
+                // phpcs:ignore Magento2.Performance.ForeachArrayMerge
```
Comment (xmav) [inline dev/tests/static/testsuite/Magento/Test/Integrity/Di/CompilerTest.php]: Could you please try to rewrite it using example from - https://github.com/magento/magento-coding-standard/issues/20

File: dev/tests/static/testsuite/Magento/Test/Integrity/Magento/Framework/Api/ExtensibleInterfacesTest.php (hunk 1 above)
Comment (xmav) [reply dev/tests/static/testsuite/Magento/Test/Integrity/Magento/Framework/Api/ExtensibleInterfacesTest.php]: Could you please revise this check to be sure it works as expected ? With this regex `/\[\s\<\w+?>\s([?]?[\w]+)/s` we will have just first part of FQCN in $matches[1]. that mean that any parameter type will be considered correct as \Magento != \Mageno\Module\Api\Data\SomeExtensionInterface

**Review by xmav — CHANGES_REQUESTED** [review]

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests with env PHP 8.0

**Review by xmav — APPROVED** [review]

**Pull request merged**